app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,26 @@
 class PostsController < ApplicationController
   def index
-    puts params
+    param_id = params['id'].to_i
+    @single_user_found = User.find_by(id: param_id)
+    @user_found_posts = Post.where(author_id: param_id)
+    @all_comments = Comment.all
   end
 
   def show
     puts params
+    param_id = params['id'].to_i
+    param_user_id = params['user_id'].to_i
+    @user_found = User.find_by(id: param_user_id)
+    @post = Post.find_by(id: param_id)
+    @all_comments = Comment.where(post_id: param_id)
+  end
+
+  def new
+    Post.new
+  end
+
+  def create
```

Comment:
- [ ] Based on the requirements of this project, a post is created on behalf of the currently logged-in user (current_user) of the app. You can simply implement it like this:



```ruby
  def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
```